algebird-core/src/main/scala/com/twitter/algebird/QTree.scala

```diff
@@ -258,6 +289,7 @@ case class QTree[A](
   }
 
   def interQuartileMean(implicit n: Numeric[A]): (Double, Double) = {
+    implicit val monoid: Monoid[A] = Ring.numericRing[A]
```

Comment:
there is actually an issue here. You can get a Monoid (or a Ring) from a Numeric, but it really does not have to be the same Monoid as you are using elsewhere. So, I guess what we really need is to take both a Numeric and a Monoid, and only use the Numeric to convert toDouble at the end.